examples/contextswitch.rs

```diff
@@ -0,0 +1,179 @@
+                .long("pid")
+                .help("Only track this pid")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("tgid")
+                .long("tgid")
+                .help("Only track this thread")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("percpu")
+                .long("percpu")
+                .help("Display context switches per cpu")
+                .takes_value(false),
+        )
+        .get_matches();
+
+    let mut sample_frequency: Option<u64> = matches
+        .value_of("sample_frequency")
+        .map(|v| v.parse().expect("Invalid sample frequency"));
+
+    let sample_period: Option<u64> = matches
+        .value_of("sample_period")
+        .map(|v| v.parse().expect("Invalid sample period"));
+
+    if !sample_frequency.is_some() && !sample_period.is_some() {
+        sample_frequency = Some(DEFAULT_SAMPLE_FREQ);
+    }
+
+    let duration: u64 = matches
+        .value_of("duration")
+        .map(|v| v.parse().expect("Invalid duration"))
+        .unwrap_or(DEFAULT_DURATION);
+
+    let mut code = include_str!("contextswitch.c").to_string();
+    code = match matches.value_of("pid") {
+        Some(pid) => code.replace("##PID_FILTER##", &format!("pid != {}", pid)),
+        _ => code.replace("##PID_FILTER##", "0"),
+    };
+    code = match matches.value_of("tgid") {
+        Some(tgid) => code.replace("##TGID_FILTER##", &format!("tgid != {}", tgid)),
+        _ => code.replace("##TGID_FILTER##", "0"),
+    };
+    if matches.is_present("percpu") {
+        code = format!("{}\n{}", "#define PERCPU", code);
+    }
+
+    let mut bpf = BPF::new(&code)?;
+    bpf.attach_perf_event(
+        "do_count",
+        PerfType::Software as u32,
+        PerfSoftwareConfig::ContextSwitches as u32,
+        sample_period,
+        sample_frequency,
+        None,
+        None,
+        None,
+    )?;
+
+    println!("Running for {} seconds", duration);
+
+    let mut durr = 0;
+    while runnable.load(Ordering::SeqCst) {
+        thread::sleep(time::Duration::new(1, 0));
+
+        if durr == duration {
+            break;
+        }
+        durr += 1;
+    }
+
+    // Count misses
+    let count_table = bpf.table("count");
+
+    if matches.is_present("percpu") {
+        println!("{:<-8} {:<-4} {:>12}", "PID", "CPU", "COUNT");
+        for entry in count_table.iter() {
+            let key = parse_struct(&entry.key);
+            let value = parse_u64(entry.value);
+
+            println!("{:<-8} {:<-4} {:>12}", key.pid, key.cpu, value);
+        }
+    } else {
+        println!("{:<-8} {:>12}", "PID", "COUNT");
+        for entry in count_table.iter() {
+            let key = parse_u32(entry.key);
+            let value = parse_u64(entry.value);
+
+            println!("{:<-8} {:>12}", key, value);
+        }
+    }
+
+    Ok(())
+}
+
+fn parse_u32(x: Vec<u8>) -> u32 {
+    let mut v = [0_u8; 4];
+    for i in 0..4 {
```

Comment:
Take a look at the changes I made in: https://github.com/rust-bpf/bpf-tools/blob/master/src/hardirqs/main.rs#L163